Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/polygon_soup_to_polygon_mesh.h

```diff
@@ -17,23 +17,99 @@
 
 #include <CGAL/disable_warnings.h>
 
-#include <CGAL/boost/graph/Euler_operations.h>
-#include <CGAL/property_map.h>
 #include <CGAL/Polygon_mesh_processing/orient_polygon_soup.h>
+
 #include <CGAL/algorithm.h>
-#include <set>
-#include <boost/dynamic_bitset.hpp>
+#include <CGAL/boost/graph/Euler_operations.h>
+#include <CGAL/boost/graph/iterator.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+#include <CGAL/Dynamic_property_map.h>
+#include <CGAL/property_map.h>
 
+#include <boost/dynamic_bitset.hpp>
 #include <boost/range/size.hpp>
 #include <boost/range/value_type.hpp>
 #include <boost/range/reference.hpp>
 
-namespace CGAL
-{
-namespace Polygon_mesh_processing
+#include <set>
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+namespace internal {
+
+// \ingroup PMP_repairing_grp
+//
+// Adds the vertices and faces of a mesh into a (possibly non-empty) soup.
+//
+// @tparam PolygonMesh a model of `FaceListGraph` with an internal point property map
+// @tparam PointRange a model of the concepts `RandomAccessContainer` and
+//                    `BackInsertionSequence` whose value type is the point type
+// @tparam PolygonRange a model of the concepts `RandomAccessContainer` and `BackInsertionSequence` whose
+//                      `value_type` is itself a model of the concepts `RandomAccessContainer` and
+//                      `BackInsertionSequence` whose `value_type` is `std::size_t`.
+//
+// @param mesh the mesh whose faces are being put in the soup
+// @param points points of the soup of polygons
+// @param polygons each element in the vector describes a polygon using the index of the points in `points`
+//
+// \sa `CGAL::Polygon_mesh_processing::orient_polygon_soup()`
+// \sa `CGAL::Polygon_mesh_processing::is_polygon_soup_a_polygon_mesh()`
+// \sa `CGAL::Polygon_mesh_processing::polygon_soup_to_polygon_mesh()`
+//
+template<typename PolygonMesh, typename PointRange, typename PolygonRange, typename NamedParameters>
+void polygon_mesh_to_polygon_soup(const PolygonMesh& mesh,
```

Comment:
Todo replace by the proper function once it is integrated (https://github.com/CGAL/cgal/pull/4418)